util/topgen/c.py

```diff
@@ -48,17 +48,24 @@ def as_c_type(self):
 
 
 class MemoryRegion(object):
-    def __init__(self, name, base_addr, size_bytes):
+    def __init__(self, name, addr, size_bytes):
         self.name = name
-        self.base_addr = base_addr
+        self.addr = addr
         self.size_bytes = size_bytes
+        self.size_words = str(int(size_bytes, 0) // 4)
```

Comment:
Don't `str` this expression, `size_bytes // 4` should be enough, especially if you don't quote the value in the hjson. The second argument of `int(<string>, <n>)` is actually a base, which I doubt you intend to be base zero.

Reply:
> ... especially if you don't quote the value in the hjson.

We have a style guide that [explicitly tells us to quote values in hjson files](https://github.com/lowRISC/opentitan/blob/master/doc/rm/hjson_usage_style.md#simple-key-value-entries). This seems slightly bonkers to me, but apparently is what the OT team has settled on. I wonder whether this can be changed.

Reply:
@rswarbrick, I agree. I think that part of the style guide should be revisited, because I don't see any issues with the second example: it doesn't use quoteless strings and `tag` is parsed as 2 as expected. AFAIK, quoteless strings require one value per line w/o commas.

Reply:
Done.

Reply:
> The second argument of int(<string>, <n>) is actually a base, which I doubt you intend to be base zero.

If I don't have `0`, then it cannot convert, and I need to convert it to integer, otherwise the CI falls over. If you know the better solution let me know, meanwhile I will `self.size_words = int(size_bytes, 0) // 4` as that seems to work.

Reply:
Just to clear this up: `int(foo, 0)` means "parse foo as an integer, guessing the base". So `int("0x10", 0)` parses as 16 and `int("10", 0)` parses as 10.